pkg/controller/logstorage/secrets/secret_controller.go

```diff
@@ -570,7 +570,10 @@ func (c *elasticKeyPairCollection) component(bundle certificatemanagement.Truste
 			// enabled. Instead, it will be replaced with a dummy secret that serves merely to pass ECK's validation checks.
 			rcertificatemanagement.NewKeyPairOption(c.elastic, true, c.elastic != nil && !c.elastic.UseCertificateManagement()),
 		},
-		TrustedBundle: bundle,
+
+		// We don't create a trusted bundle as part of the elastic component - this is just to install the keypair secret.
+		// The trusted bundle will be installed later, and include other upstream certificates as well.
+		TrustedBundle: nil,
```

Comment:
@tmjd I think this was effectively the problem line - we were creating a trusted bundle here that only included the tigera-ca and Linseed certificate.

Then, on  line 316 we would create another trusted bundle that included the tigera-ca, Linseed certificate, as well as many other things - critically, the external ES public certificate.

Reply:
I've changed it so that the internal ES component only creates KeyPairs in the tiger-elasticsearch Namespace (it still creates a bundle in the tigera-kibana namespace, though).

Then, a single trusted bundle is created shortly after, avoiding having two fighting representations of what should be in it.

Reply:
I might be able to simplify this even further by just getting rid of these separate keyPair components and doing it all in one place, but that's a larger refactor.

Most of this logic was added originally because we were testing multi-tenancy against an internal ES. But, since that's not actually a supported mode of operation, a lot of the hoops I had to jump through can now be removed. Hooray!